File: scheduledTasks/dailyGameGenerator.js
const fs = require('fs');

const INDICES = [0, 1, 2, 3, 4, 5];

let data = {};
let easyArray = [];
let mediumArray = [];
let hardArray = [];

// answer contains 4 possible colors, without duplicates
function easyAnswer() {
  let answer = [0, 2, 3, 4];
  for (let i = answer.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [answer[i], answer[j]] = [answer[j], answer[i]];
  }

  return answer;
}
// answer contains 6 possible colors, without duplicates
function mediumAnswer() {
  let answer = [];

  while (answer.length < 4) {
    const randomColor = INDICES[
      Math.floor(Math.random() * INDICES.length)
    ];

    if (!answer.includes(randomColor)) {
      answer.push(randomColor);
    }
  }

  return answer;
}
// answer contains 6 possible colors, and may include duplicates
function hardAnswer() {
  let answer = [];

  for (let i = 0; i < 4; i++) {
      answer.push(
      INDICES[
          Math.floor(Math.random() * INDICES.length)
      ]
      );
  }

  return answer;
}
// load array with 256 unique easy answers
function loadEasyArray() {
  for (let i = 0; i < 256; i++) {
    let answer = easyAnswer();
    while (easyArray.includes(answer)) {
      answer = easyAnswer();
    }
    easyArray.push(easyAnswer());
  }
}
// load array with 360 unique medium answers
function loadMediumArray() {
  for (let i = 0; i < 360; i++) {
    let answer = mediumAnswer();
    while (mediumArray.includes(answer)) {
      answer = mediumAnswer();
    }
    mediumArray.push(mediumAnswer());
  }
}
// load array with 366 unique hard answers
function loadHardArray() {
  for (let i = 0; i < 366; i++) {
    let answer = hardAnswer();
    while (hardArray.includes(answer)) {
      answer = hardAnswer();
    }
    hardArray.push(hardAnswer());
  }
}

function loadDailyGames() {
  // load arrays with unique answers
  loadEasyArray();
  loadMediumArray();
  loadHardArray();

  for (let month = 1; month <= 12; month++) {
    for (let day = 1; day <= 31; day++) {
      if (month === 2 && day > 29) {
        break;
      } else if ((month === 4 || month === 6 || month === 9 || month === 11) && day > 30) {
        break;
      } else {
        const key = `${month.toString().padStart(2, '0')}-${day.toString().padStart(2, '0')}`;
        data[key] = {
          Easy: easyArray.pop(),
          Medium: mediumArray.pop(),
          Hard: hardArray.pop()
        };
        // reload arrays if they are empty
        if (easyArray.length === 0) {
          loadEasyArray();
        }
        if (mediumArray.length === 0) {
          loadMediumArray();
        }
      }
    }
  }

  fs.writeFile('src/dailyGames.json', JSON.stringify(data, null, 2), (err) => {
    if (err) throw err;
    console.log('Data written to file');
  });
}

loadDailyGames();

File: scheduledTasks/daily_game_generator.rb
require 'date'
require 'json'

class DailyGameGenerator
  # load the daily game data
  def load_daily_game
    data = {}
    easy_array = load_answer_array(24, method(:easy_answer))
    medium_array = load_answer_array(360, method(:medium_answer))
    hard_array = load_answer_array(366, method(:hard_answer))

    start_date = Date.new(Date.today.year, 1, 1)
    end_date = start_date.next_year

    (start_date...end_date).each do |date|
      date_string = date.strftime('%m-%d-%Y')
      data[date_string] = {
        Easy: easy_array.pop,
        Medium: medium_array.pop,
        Hard: hard_array.pop
      }

      easy_array = load_answer_array(24, method(:easy_answer)) if easy_array.empty?
      medium_array = load_answer_array(360, method(:medium_answer)) if medium_array.empty?
    end

    begin
      File.open("src/dailyGames.json","w") do |f|
        f.write(JSON.pretty_generate(data))
      end
    rescue => e
      puts "An error occurred: #{e.message}"
    end
  end
  # load the answer arrays; accept array count and answer generator as arguments
  def load_answer_array(count, get_answer_method)
    answer_array = []

    count.times do
      answer = get_answer_method.call
      while answer_array.include?(answer)
        answer = get_answer_method.call
      end
      answer_array.push(answer)
    end

    answer_array
  end
  # answer contains 4 possible colors, without duplicates
  def easy_answer
    answer = [0, 2, 3, 4]
    answer.shuffle!
  end
  # answer contains 6 possible colors, without duplicates
  def medium_answer
    answer = (0..5).to_a
    answer.shuffle!
    answer.pop
    answer.pop
    answer
  end
  # answer contains 6 possible colors, and may include duplicates
  def hard_answer
    answer = []
    4.times do
      answer.push(rand(0..5))
    end
    answer
  end
end



DailyGameGenerator.new.load_daily_game